src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,135 @@
+import duke.ui.Ui;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class AddCommand extends Command {
+    private Task task;
+    private String type;
+    private String description;
+    private String time;
+
+    /**
+     * Initialises an AddCommand.
+     *
+     * @param type        The type of AddCommand
+     * @param description The description of the command
+     * @throws DukeException In the event of empty descriptions/invalid commands/ invalid command formats.
+     */
+    public AddCommand(String type, String[] description) throws DukeException {
+
+        try {
+            String[] details;
+            this.description = description[1].trim();
+            this.type = type;
+            switch (type) {
+            case "todo":
+                task = new ToDo(parseDescription());
+                break;
+            case "deadline":
+                task = new Deadline(parseDescription("/by"), parseDate());
+                break;
+            case "event":
+                task = new Event(parseDescription("/at"), parseDate());
+                break;
+            default:
+                break;
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new DukeException("The description of a " + type + " cannot be empty");
+        }
+    }
+
+    /**
+     * Executes the AddCommand, adds the task to the LinkedList and saves it to the file.
+     *
+     * @param tasks   The TaskList containing all existing tasks.
+     * @param ui      The Ui for printing purposes.
+     * @param storage The Storage for saving tasks to file.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+
+        tasks.allTasks.add(task);
+        StringBuilder sb = new StringBuilder("Got it. I've added this task:\n" + task);
+        if (tasks.allTasks.size() == 1) {
+            sb.append("\nNow you have ")
+              .append(tasks.allTasks.size())
+                .append(" task in the list.");
+        } else {
+            sb.append("\nNow you have ")
+              .append(tasks.allTasks.size())
+                .append(" tasks in the list.");
+        }
+        storage.appendTaskToFile(task);
+        ui.printMessage(sb.toString());
+    }
+
+    /**
+     * Checks if it is a bye command.
+     *
+     * @return A boolean: true if it is a bye command.
+     */
+    public boolean isExit() {
+        return false;
+    }
+
+    /**
+     * Parses a string and reformats it as a date.
+     *
+     * @return The string representation of the new date object.
+     */
+    public String parseDate() {
+        try {
+            SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy HHmm", java.util.Locale.ENGLISH);
+            Date myDate = sdf.parse(time.replaceAll("-", "/"));
+            sdf.applyPattern("EEE, d MMM yyyy, hh:mm a");
+            return sdf.format(myDate);
+        } catch (ParseException e) {
```

Comment:
Good effort in parsing the date and applying a different pattern to format to another date format that may be more readable.